### ISSUE TITLE
ensure some metadata tests for all linters

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2221]
I noticed we have no tests for metadata (location).
Feel free to add some or merge as-is.

_Originally posted by @AshesITR in https://github.com/r-lib/lintr/pull/2186#pullrequestreview-1652468495_

actually I think this is a more general concern, we should make a sweep ensuring all linters have some test of metadata

----

As of now, the following linters lack any metadata tests:

 - [ ] absolute_path_linter
 - [ ] any_duplicated_linter
 - [ ] any_is_na_linter
 - [ ] backport_linter
 - [ ] boolean_arithmetic_linter
 - [ ] class_equals_linter
 - [ ] commas_linter
 - [ ] condition_call_linter
 - [ ] condition_message_linter
 - [ ] conjunct_test_linter
 - [ ] consecutive_assertion_linter
 - [ ] consecutive_mutate_linter
 - [ ] cyclocomp_linter
 - [ ] duplicate_argument_linter
 - [ ] empty_assignment_linter
 - [ ] expect_comparison_linter
 - [ ] expect_identical_linter
 - [ ] expect_length_linter
 - [ ] expect_named_linter
 - [ ] expect_not_linter
 - [ ] expect_null_linter
 - [ ] expect_s3_class_linter
 - [ ] expect_s4_class_linter
 - [ ] expect_true_false_linter
 - [ ] expect_type_linter
 - [ ] fixed_regex_linter
 - [ ] for_loop_index_linter
 - [ ] function_return_linter
 - [ ] ifelse_censor_linter
 - [ ] indentation_linter
 - [ ] infix_spaces_linter
 - [ ] inner_combine_linter
 - [ ] is_numeric_linter
 - [ ] length_levels_linter
 - [ ] length_test_linter
 - [ ] lengths_linter
 - [ ] line_length_linter
 - [ ] literal_coercion_linter
 - [ ] matrix_apply_linter
 - [ ] missing_argument_linter
 - [ ] missing_package_linter
 - [ ] namespace_linter
 - [ ] nested_ifelse_linter
 - [ ] nonportable_path_linter
 - [ ] numeric_leading_zero_linter
 - [ ] outer_negation_linter
 - [ ] package_hooks_linter
 - [ ] pipe_consistency_linter
 - [ ] pipe_return_linter
 - [ ] quotes_linter
 - [ ] redundant_equals_linter
 - [ ] redundant_ifelse_linter
 - [ ] regex_subset_linter
 - [ ] routine_registration_linter
 - [ ] scalar_in_linter
 - [ ] seq_linter
 - [ ] sort_linter
 - [ ] spaces_left_parentheses_linter
 - [ ] sprintf_linter
 - [ ] strings_as_factors_linter
 - [ ] system_file_linter
 - [ ] todo_comment_linter
 - [ ] trailing_blank_lines_linter
 - [ ] trailing_whitespace_linter
 - [ ] unnecessary_lambda_linter
 - [ ] unnecessary_nesting_linter
 - [ ] unnecessary_placeholder_linter
 - [ ] unused_import_linter
 - [ ] vector_logic_linter
 - [ ] whitespace_linter
 - [ ] yoda_test_linter

**Comment by MichaelChirico** [comment]
Working on a script for this.

 1. Edit `xml_nodes_to_lints()` to always report `line_number = as.integer(2^31-1)`
 2. Run the following with testthat:

```r
library(testthat)
library(lintr)

report <- test_dir(
  "tests/testthat",
  filter = "linter$",
  stop_on_failure=FALSE,
  reporter = SilentReporter$new()
)
names(report) <- sapply(report, `[[`, "file")

failed <- report |>
  sapply(\(x) any(sapply(x$results, inherits, "expectation_failure"))) |>
  which() |>
  names() |>
  unique() |>
  gsub(pattern = "^test-|\\.R$", replacement = "")

passed <- setdiff(
  available_linters(tags = NULL)$linter,
  failed
)
```

Added the list of linters that need tests to the issue body. It misses linters not using `xml_nodes_to_lints()` (I guess there are not many of these but haven't checked) but gets us most of the way there.

**Comment by AshesITR** [comment]
Any reason to not edit `Lint()`?

**Comment by MichaelChirico** [comment]
> Any reason to not edit `Lint()`?

no, that's a much better idea.
